### PR TITLE
Fix for the issue #1714

### PR DESCRIPTION
**Pull request by Madhu-NEC**
Updated MongoBackendImpl.java for [issue#1714](https://github.com/telefonicaid/fiware-cygnus/issues/1714#issuecomment-538975997)

### CODE REVIEW

File: cygnus-common/src/main/java/com/telefonica/iot/cygnus/backends/mongo/MongoBackendImpl.java [hunk 1]
```diff
@@ -124,7 +124,7 @@ public void createCollection(String dbName, String collectionName, long dataExpi
                 .append("_id.attrName", 1)
                 .append("_id.resolution", 1)
                 .append("_id.origin", 1);
-            options = new IndexOptions();
+            options = new IndexOptions().name("index");
```
Comment (fgalan) [inline cygnus-common/src/main/java/com/telefonica/iot/cygnus/backends/mongo/MongoBackendImpl.java]:
```suggestion
            options = new IndexOptions().name("cygnus_agg_optimization_index");
```

In addition, the index created a L136 should have a different name, e.g: "cygnus_agg_expiration_index"

File: cygnus-common/src/main/java/com/telefonica/iot/cygnus/backends/mongo/MongoBackendImpl.java [hunk 2]
```diff
@@ -185,7 +185,7 @@ public void createCollection(String dbName, String collectionName, long collecti
                 .append("entityType", 1)
                 .append("attrName", 1)
                 .append("recvTime", 1);
-            options = new IndexOptions();
+            options = new IndexOptions().name("index");
```
Comment (fgalan) [inline cygnus-common/src/main/java/com/telefonica/iot/cygnus/backends/mongo/MongoBackendImpl.java]:
```suggestion
            options = new IndexOptions().name("cygnus_raw_optimization_index");
```

In addition, the index created a L197 should have a different name, e.g: "cygnus_raw_expiration_index"

File: CHANGES_NEXT_RELEASE [hunk 3]
```diff
@@ -1,2 +1,3 @@
 [cygnus-common][PostgreSQL][MySQL] catch exception close invalid connection
 [cygnus-common] Upgrade MongoDB driver from 3.0.4 to 3.11.0
+[cygnus-common] Fix for index name creation in Mongo Sink (#1714)
```
Comment (fgalan) [inline CHANGES_NEXT_RELEASE]:
```suggestion
[cygnus-common] Fix for index name creation in Mongo and STH sinks (#1714)
```

**Comment by Madhu-NEC** [comment]
Hi @fgalan,

As mentioned in [mongo document](https://docs.mongodb.com/manual/reference/limits/#namespaces), the limit for collection namespace `<database>.<collection>` is 120 bytes and [index key limit](https://docs.mongodb.com/manual/reference/limits/#Index-Name-Length) `<database name>.<collection name>.$<index name>` is 127 bytes. 
As per my understanding the index name should be of 5 bytes or less, so that it can allow 120 bytes of collection namespace. 

For example, index name given as "cygnus_raw_expiration_index" that means 27 bytes.
**Case 1: If collection namespace is less than or equal to 98 bytes.**
Cygnus will create index successfully.

**Case 2: If collection namespace is greater than 98 bytes and less than 120 bytes.**
Cygnus will not create index as index key limit will exceed 127 bytes `<database>.<collection>=110` + `<index name>=27` + 2 (for `.$`) =139 bytes

I have concern about Case 2 as index creation is failing in this case.

**Could you please provide your opinion?**

**Comment by fgalan** [comment]
Good point. Maybe 5 bytes is too short, but my original proposals could be shortened. For instance:

cyg_agg_exp
cyg_agg_opt
cyg_raw_exp
cyg_raw_opt

File: cygnus-common/src/main/java/com/telefonica/iot/cygnus/backends/mongo/MongoBackendImpl.java (hunk 2 above)
Comment (fgalan) [reply cygnus-common/src/main/java/com/telefonica/iot/cygnus/backends/mongo/MongoBackendImpl.java]: Fixed in 64d4802

File: CHANGES_NEXT_RELEASE (hunk 3 above)
Comment (fgalan) [reply CHANGES_NEXT_RELEASE]: Fixed in 64d4802

**Pull request merged**